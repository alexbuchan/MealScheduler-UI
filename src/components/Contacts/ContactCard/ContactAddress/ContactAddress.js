import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  item: PropTypes.arrayOf(PropTypes.string) // The identifier and value for parts of an address
};

const ContactAddress = ({ index, item }) => {
  const label = item[0].replace(/^\w/, c => c.toUpperCase());

  return (
    <li className="contact-item address-item" ><strong>{ label }</strong>: { item[1] }</li>
  );
}

ContactAddress.propTypes = propTypes;
export default ContactAddress;