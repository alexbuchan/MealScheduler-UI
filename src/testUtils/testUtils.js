import React from 'react';
import { shallow } from 'enzyme';

export const componentSetup = (Component, props = {}, options = {}) => {
  const instance = shallow(<Component { ...props } />, options);
  return instance;
}

export const findByTestAttribute = (instance, attribute) => {
  const wrapper = instance.find(`[data-test="${attribute}"]`);
  return wrapper;
}