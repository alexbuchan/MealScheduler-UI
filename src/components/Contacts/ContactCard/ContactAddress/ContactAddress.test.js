import ContactAddress from './ContactAddress';
import { componentSetup } from '../../../../testUtils/testUtils';

describe('ContactAddress', () => {
  let instance;
  let props;
  
  beforeEach(() => {
    props = {
      index: 0,
      item: ['Street', 'Kulas Light']
    };

    instance = componentSetup(ContactAddress, props);
  });

  it('Should render without errors', () => {
    expect(instance).toMatchSnapshot();
  });
});