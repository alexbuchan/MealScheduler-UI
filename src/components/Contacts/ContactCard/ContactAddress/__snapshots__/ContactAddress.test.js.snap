// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAddress Should render without errors 1`] = `
<li
  className="contact-item address-item"
>
  <strong>
    Street
  </strong>
  : 
  Kulas Light
</li>
`;
